apps/emqx_auth_http/src/emqx_authn_scram_http.erl

```diff
@@ -0,0 +1,179 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2024 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%--------------------------------------------------------------------
+
+-module(emqx_authn_scram_http).
+
+-include_lib("emqx_auth/include/emqx_authn.hrl").
+-include_lib("emqx/include/logger.hrl").
+
+-behaviour(emqx_authn_provider).
+
+-export([
+    create/2,
+    update/2,
+    authenticate/2,
+    destroy/1
+]).
+
+-define(REQUIRED_USER_INFO_KEYS, [
+    <<"stored_key">>,
+    <<"server_key">>,
+    <<"salt">>
+]).
+
+-define(OPTIONAL_USER_INFO_KEYS, [
```

Comment:
A separate (maybe low prio) PR after this is merged to `master`:
add `<<"acl">>` here, also add some tests to verify ACL checks.

Reply:
it's actually already in release-57: https://github.com/emqx/emqx/pull/13400